From f950e4773ac12fd5c7718e899016133ea68c40c8 Mon Sep 17 00:00:00 2001
From: Alejandro Sierra <b18039@freescale.com>
Date: Thu, 31 Mar 2016 14:20:32 -0500
Subject: [PATCH 21/29] Add imx6dqscm ePoP support

Add imx6dqscm ePoP support. This support was added for
the following boards.
    - IMX6DQSCM HVB board
    - IMX6DQSCM EVB board
    - IMX6DQSCM QWKS rev2 board

Signed-off-by: Alejandro Sierra <b18039@freescale.com>
---
 .../freescale/mx6dqscmqwks/imximage_scm_lpddr2.cfg | 150 ++++++++++++++++++++-
 board/freescale/mx6dqscmqwks/mx6dqscmqwks.c        |   2 +-
 configs/mx6dqscm_1gb_interleaving_hvb_defconfig    |   5 +
 configs/mx6dqscm_512mb_evb_defconfig               |   5 +
 configs/mx6dqscm_512mb_hvb_defconfig               |   5 +
 configs/mx6dqscm_512mb_qwks_rev2_defconfig         |   5 +
 configs/mx6dqscm_512mb_qwks_rev2_spinor_defconfig  |   7 +
 include/configs/mx6dqscmqwks.h                     |   9 +-
 8 files changed, 184 insertions(+), 4 deletions(-)
 create mode 100644 configs/mx6dqscm_1gb_interleaving_hvb_defconfig
 create mode 100644 configs/mx6dqscm_512mb_evb_defconfig
 create mode 100644 configs/mx6dqscm_512mb_hvb_defconfig
 create mode 100644 configs/mx6dqscm_512mb_qwks_rev2_defconfig
 create mode 100644 configs/mx6dqscm_512mb_qwks_rev2_spinor_defconfig

diff --git a/board/freescale/mx6dqscmqwks/imximage_scm_lpddr2.cfg b/board/freescale/mx6dqscmqwks/imximage_scm_lpddr2.cfg
index 77cd459..bb3d542 100755
--- a/board/freescale/mx6dqscmqwks/imximage_scm_lpddr2.cfg
+++ b/board/freescale/mx6dqscmqwks/imximage_scm_lpddr2.cfg
@@ -44,7 +44,155 @@ CSF 0x2000
  *	Address	  absolute address of the register
  *	value	  value to be stored in the register
  */
-#ifdef CONFIG_SCM_LPDDR2_2GB
+#ifdef CONFIG_SCM_LPDDR2_512MB
+/* DCD */
+/* DDR clock to 400MHz */
+DATA 4, 0x020C4018 0x00060324 
+/* AHB_ROOT_CLK change divide ratio from 4 to 3 for ENET */
+DATA 4, 0x020C4014 0x00018900 
+
+DATA 4 0x020C4018 0x00060324
+
+DATA 4 0x020e0798 0x00080000
+DATA 4 0x020e0758 0x00000000
+
+DATA 4 0x020E0588 0x00000020
+DATA 4 0x020E0594 0x00000020
+
+DATA 4 0x020E056c 0x00000020
+DATA 4 0x020E0578 0x00000020
+DATA 4 0x020E074c 0x00000020
+
+DATA 4 0x020E057c 0x00000020
+DATA 4 0x020E058c 0x00000000
+DATA 4 0x020E059c 0x00000020
+DATA 4 0x020E05a0 0x00000020
+DATA 4 0x020E078c 0x00000020
+
+DATA 4 0x020E0750 0x00020000
+DATA 4 0x020E05a8 0x00003020
+DATA 4 0x020E05b0 0x00003020
+DATA 4 0x020E0524 0x00003020
+DATA 4 0x020E051c 0x00003020
+DATA 4 0x020E0518 0x00003020
+DATA 4 0x020E050c 0x00003020
+DATA 4 0x020E05b8 0x00003020
+DATA 4 0x020E05c0 0x00003020
+
+DATA 4 0x020E0774 0x00020000
+
+DATA 4 0x020E0784 0x00000020
+DATA 4 0x020E0788 0x00000020
+DATA 4 0x020E0794 0x00000020
+DATA 4 0x020E079c 0x00000020
+DATA 4 0x020E07a0 0x00000020
+DATA 4 0x020E07a4 0x00000020
+DATA 4 0x020E07a8 0x00000020
+DATA 4 0x020E0748 0x00000020
+
+DATA 4 0x020E05ac 0x00000020
+DATA 4 0x020E05b4 0x00000020
+DATA 4 0x020E0528 0x00000020
+DATA 4 0x020E0520 0x00000020
+DATA 4 0x020E0514 0x00000020
+DATA 4 0x020E0510 0x00000020
+DATA 4 0x020E05bc 0x00000020
+DATA 4 0x020E05c4 0x00000020
+
+DATA 4 0x020E0590 0x00000020
+DATA 4 0x020E0598 0x00000020
+
+DATA 4 0x021b001c 0x00008000
+
+DATA 4 0x021b085c 0x1b4700c7
+
+DATA 4 0x021b0800 0xa1390003
+
+
+DATA 4 0x021b0890 0x00400000
+
+DATA 4 0x021b0848 0x44404044
+
+DATA 4 0x021b0850 0x34343A38
+
+DATA 4 0x021b083c 0x20000000
+DATA 4 0x021b0840 0x00000000
+
+DATA 4 0x021b081c 0x33333333
+DATA 4 0x021b0820 0x33333333
+DATA 4 0x021b0824 0x33333333
+DATA 4 0x021b0828 0x33333333
+
+DATA 4 0x021b082c 0xf3333333
+DATA 4 0x021b0830 0xf3333333
+DATA 4 0x021b0834 0xf3333333
+DATA 4 0x021b0838 0xf3333333
+
+DATA 4 0x021b08b8 0x00000800
+
+DATA 4 0x021b0004 0x00020036
+DATA 4 0x021b0008 0x00000000
+DATA 4 0x021b000c 0x33374133
+
+DATA 4 0x021b0010 0x00100a82
+
+DATA 4 0x021b0014 0x00000093
+
+DATA 4 0x021b0018 0x0000174C
+DATA 4 0x021b001c 0x00008000
+DATA 4 0x021b002c 0x0f9f26d2
+DATA 4 0x021b0030 0x00000010
+DATA 4 0x021b0038 0x00190778
+
+//FIX MODE
+DATA 4 0x021b0040 0x0000004f
+
+DATA 4 0x021b0400 0x11420000
+
+DATA 4 0x021b0000 0x83110000
+
+
+// Channel 0 
+DATA 4 0x021b001c 0x003f8030
+DATA 4 0x021b001c 0xff0a8030
+DATA 4 0x021b001c 0x82018030
+DATA 4 0x021b001c 0x04028030
+DATA 4 0x021b001c 0x04038030
+
+// Channel 1 
+//DATA 4 0x021b401c 0x003f8030
+//DATA 4 0x021b401c 0xff0a8030
+//DATA 4 0x021b401c 0x82018030
+//DATA 4 0x021b401c 0x04028030
+//DATA 4 0x021b401c 0x04038030
+
+DATA 4 0x021b0800 0xa1390003
+
+DATA 4 0x021b0020 0x00001800
+
+DATA 4 0x021b0818 0x00000000
+
+DATA 4 0x021b0004 0x00025576
+
+DATA 4 0x021b0404 0x00011006
+
+DATA 4 0x021b001c 0x00000000
+
+
+DATA 4 0x020c4068 0x00C03F3F
+DATA 4 0x020c406c 0x0030FC03
+DATA 4 0x020c4070 0x0FFFC000
+DATA 4 0x020c4074 0x3FF00000
+DATA 4 0x020c4078 0x00FFF300
+DATA 4 0x020c407c 0x0F0000C3
+DATA 4 0x020c4080 0x000003FF
+
+/* enable AXI cache for VDOA/VPU/IPU */
+DATA 4 0x020e0010 0xF00000CF
+/* set IPU AXI-id0 Qos=0xf(bypass) AXI-id1 Qos=0x7 */
+DATA 4 0x020e0018 0x007F007F
+DATA 4 0x020e001c 0x007F007F
+#elif CONFIG_SCM_LPDDR2_2GB
 /* DDR clock to 400MHz */
 DATA 4, 0x020C4018 0x00060324
 /* AHB_ROOT_CLK change divide ratio from 4 to 3 for ENET */
diff --git a/board/freescale/mx6dqscmqwks/mx6dqscmqwks.c b/board/freescale/mx6dqscmqwks/mx6dqscmqwks.c
index 9a0a010..8047ae7 100644
--- a/board/freescale/mx6dqscmqwks/mx6dqscmqwks.c
+++ b/board/freescale/mx6dqscmqwks/mx6dqscmqwks.c
@@ -105,7 +105,7 @@ int dram_init(void)
 	return 0;
 }
 
-#ifndef CONFIG_INTERLEAVING_MODE
+#if !defined(CONFIG_INTERLEAVING_MODE) && !defined(CONFIG_SCM_LPDDR2_512MB)
 void dram_init_banksize(void)
 {
        gd->bd->bi_dram[0].start = PHYS_SDRAM_0;
diff --git a/configs/mx6dqscm_1gb_interleaving_hvb_defconfig b/configs/mx6dqscm_1gb_interleaving_hvb_defconfig
new file mode 100644
index 0000000..be56cd1
--- /dev/null
+++ b/configs/mx6dqscm_1gb_interleaving_hvb_defconfig
@@ -0,0 +1,5 @@
+CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6dqscmqwks/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=1024,SYS_USE_SPINOR,DEFAULT_FDT_FILE=\"imx6dqscm-1gb-hvb-fix-ldo.dtb\",INTERLEAVING_MODE,SCMHVB"
+CONFIG_ARM=y
+CONFIG_TARGET_MX6DQSCMQWKS=y
+CONFIG_DM=y
+CONFIG_DM_THERMAL=y
diff --git a/configs/mx6dqscm_512mb_evb_defconfig b/configs/mx6dqscm_512mb_evb_defconfig
new file mode 100644
index 0000000..6c26123
--- /dev/null
+++ b/configs/mx6dqscm_512mb_evb_defconfig
@@ -0,0 +1,5 @@
+CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6dqscmqwks/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=512,SYS_USE_SPINOR,DEFAULT_FDT_FILE=\"imx6dqscm-512mb-evb-fix-ldo.dtb\",SCMEVB,SCM_LPDDR2_512MB"
+CONFIG_ARM=y
+CONFIG_TARGET_MX6DQSCMQWKS=y
+CONFIG_DM=y
+CONFIG_DM_THERMAL=y
diff --git a/configs/mx6dqscm_512mb_hvb_defconfig b/configs/mx6dqscm_512mb_hvb_defconfig
new file mode 100644
index 0000000..9726fcc
--- /dev/null
+++ b/configs/mx6dqscm_512mb_hvb_defconfig
@@ -0,0 +1,5 @@
+CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6dqscmqwks/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=512,SYS_USE_SPINOR,DEFAULT_FDT_FILE=\"imx6dqscm-512mb-hvb-fix-ldo.dtb\",SCMHVB,SCM_LPDDR2_512MB"
+CONFIG_ARM=y
+CONFIG_TARGET_MX6DQSCMQWKS=y
+CONFIG_DM=y
+CONFIG_DM_THERMAL=y
diff --git a/configs/mx6dqscm_512mb_qwks_rev2_defconfig b/configs/mx6dqscm_512mb_qwks_rev2_defconfig
new file mode 100644
index 0000000..781861d
--- /dev/null
+++ b/configs/mx6dqscm_512mb_qwks_rev2_defconfig
@@ -0,0 +1,5 @@
+CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6dqscmqwks/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=512,SYS_USE_SPINOR,DEFAULT_FDT_FILE=\"imx6dqscm-512mb-qwks-rev2-fix-ldo.dtb\",SCM_LPDDR2_512MB"
+CONFIG_ARM=y
+CONFIG_TARGET_MX6DQSCMQWKS=y
+CONFIG_DM=y
+CONFIG_DM_THERMAL=y
diff --git a/configs/mx6dqscm_512mb_qwks_rev2_spinor_defconfig b/configs/mx6dqscm_512mb_qwks_rev2_spinor_defconfig
new file mode 100644
index 0000000..0e58c46
--- /dev/null
+++ b/configs/mx6dqscm_512mb_qwks_rev2_spinor_defconfig
@@ -0,0 +1,7 @@
+CONFIG_SYS_EXTRA_OPTIONS="IMX_CONFIG=board/freescale/mx6dqscmqwks/imximage_scm_lpddr2.cfg,MX6Q,DDR_MB=512,SYS_BOOT_SPINOR,DEFAULT_FDT_FILE=\"imx6dqscm-512mb-qwks-rev2-fix-ldo.dtb\",SCM_LPDDR2_512MB"
+CONFIG_ARM=y
+CONFIG_TARGET_MX6DQSCMQWKS=y
+CONFIG_SYS_MALLOC_F=y
+CONFIG_SYS_MALLOC_F_LEN=0x400
+CONFIG_DM=y
+CONFIG_DM_THERMAL=y
diff --git a/include/configs/mx6dqscmqwks.h b/include/configs/mx6dqscmqwks.h
index e1babdc..3ce9360 100644
--- a/include/configs/mx6dqscmqwks.h
+++ b/include/configs/mx6dqscmqwks.h
@@ -27,7 +27,12 @@
 #endif
 #define CONFIG_MMCROOT			"/dev/mmcblk2p2"  /* SDHC3 */
 #define CONFIG_DMFC				"dmfc=3"
-#define PHYS_SDRAM_SIZE		(1u * 1024 * 1024 * 1024)
+
+#ifdef CONFIG_SCM_LPDDR2_512MB
+#define PHYS_SDRAM_SIZE         (512 * 1024 * 1024)
+#else
+#define PHYS_SDRAM_SIZE         (1u * 1024 * 1024 * 1024)
+#endif
 
 #include "mx6sabre_common.h"
 
@@ -40,7 +45,7 @@
 #endif
 
 /* Physical Memory Map */
-#ifndef CONFIG_INTERLEAVING_MODE
+#if !defined(CONFIG_INTERLEAVING_MODE) && !defined(CONFIG_SCM_LPDDR2_512MB)
 #undef CONFIG_NR_DRAM_BANKS
 #define CONFIG_NR_DRAM_BANKS            2
 #ifdef CONFIG_SCM_LPDDR2_2GB
-- 
2.7.4

